src/test/javascript/spec/component/learning-paths/participate/learning-path-lecture-unit-view.component.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MockModule } from 'ng-mocks';
+import { ArtemisTestModule } from '../../../test.module';
+import { RouterModule } from '@angular/router';
+import { Lecture } from 'app/entities/lecture.model';
+import { LearningPathLectureUnitViewComponent, LectureUnitCompletionEvent } from 'app/course/learning-paths/participate/lecture-unit/learning-path-lecture-unit-view.component';
+import { AttachmentUnit } from 'app/entities/lecture-unit/attachmentUnit.model';
+import { ArtemisLectureUnitsModule } from 'app/overview/course-lectures/lecture-units.module';
+import { LectureUnitService } from 'app/lecture/lecture-unit/lecture-unit-management/lectureUnit.service';
+import { VideoUnit } from 'app/entities/lecture-unit/videoUnit.model';
+import { TextUnit } from 'app/entities/lecture-unit/textUnit.model';
+import { OnlineUnit } from 'app/entities/lecture-unit/onlineUnit.model';
+import { Course, CourseInformationSharingConfiguration } from 'app/entities/course.model';
+import { DiscussionSectionComponent } from 'app/overview/discussion-section/discussion-section.component';
+
+describe('LearningPathLectureUnitViewComponent', () => {
+    let fixture: ComponentFixture<LearningPathLectureUnitViewComponent>;
+    let comp: LearningPathLectureUnitViewComponent;
+    let lecture: Lecture;
+    let lectureUnitService: LectureUnitService;
+    let setCompletionStub: jest.SpyInstance;
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, MockModule(RouterModule), MockModule(ArtemisLectureUnitsModule)],
+            declarations: [LearningPathLectureUnitViewComponent],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(LearningPathLectureUnitViewComponent);
+                comp = fixture.componentInstance;
+                lecture = new Lecture();
+                lecture.id = 1;
+                lectureUnitService = TestBed.inject(LectureUnitService);
+                setCompletionStub = jest.spyOn(lectureUnitService, 'setCompletion');
+            });
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should display attachment unit correctly', () => {
```

Comment:
Have you thought about using a parameterized test instead of all the different ones?